web/client/reducers/timeline.js

```diff
@@ -72,6 +73,16 @@ module.exports = (state = {
         case MOUSE_EVENT: {
             return set('mouseEvent', action.eventData, state);
         }
+        case REMOVE_NODE: {
+            if (state.selectedLayer === action.node) {
+                let newState = state;
+                return assign({}, state, {
+                     rangeData: pickBy(newState.rangeData, (values, key) => key !== action.node),
+                     loading: pickBy(newState.loading, (values, key) => key !== action.node),
+                     selectedLayer: ''
```

Comment:
You must selectedLayer to `undefined`, an empty string is not clear and even hard to maintain. Don't use the trick that empty string is falsy, please. 

Note: Please prefer spread operator over "assign", at least in reducers.
Note: Not mandatory, you can use "spread operator" and "Computed Property Name" to override the object. here an example. Anyway also this solution is ok (except for the selectedLayer value)
```
return {
   ...state,  // this copies all the properties
    rangeData: {
          ...state.rangeData, // copy again
         [action.node]: undefined  // short way to override the value of the property 
    },
    // ... the same of range data for loading
   selectedLayer: action.node === state.selectedLayer ? undefined : state.selectedLayer
}
```